api/src/pcapi/core/providers/titelive_apply_new_filters_on_history.py

```diff
@@ -0,0 +1,345 @@
+
+    return None
+
+
+def _get_remote_to_local_cmd(file_path: str, env: str) -> str:
+    return f"kubectl cp -n {env} {HOSTNAME}:{file_path} {os.path.basename(file_path)}"
+
+
+def update_products(file_path: str, comments_file_path: str, dry: bool) -> None:
+    product_whitelist_eans = {
+        ean for ean, in fraud_models.ProductWhitelist.query.with_entities(fraud_models.ProductWhitelist.ean).all()
+    }
+
+    comments_dict: dict[str, str] = {}
+
+    with open(comments_file_path, newline="", encoding="utf-8") as csv_comments_file:
+        csv_reader = csv.reader(csv_comments_file, delimiter=";")
+        for index, row in enumerate(csv_reader, HEADER_LINE_COUNT):  # pylint:disable=unused-variable
+            ean = row[COMMENT_COLUMN_INDICES["EAN"]]
+            if ean:
+                comments_dict[ean] = row[COMMENT_COLUMN_INDICES["COMMENT"]]
+
+    with open(file_path, newline="", encoding="utf-8") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=";")
+
+        total_rows = 0
+
+        whitelisted_eans: list[str] = []
+        rejected_eans: list[str] = []
+        reasons: dict[str, dict] = collections.defaultdict(lambda: {"eans": [], "total": 0})
+        emails: list[str] = []
+
+        for index, row in enumerate(csv_reader, HEADER_LINE_COUNT):
+            total_rows += 1
+            ean = row[COLUMN_INDICES["EAN"]]
+            gtl_id = row[COLUMN_INDICES["CODE_GTL"]]
+            code_support = row[COLUMN_INDICES["CODE_SUPPORT"]]
+            id_lectorat = row[COLUMN_INDICES["IDLECTORAT"]]
+            taux_tva = row[COLUMN_INDICES["TAUX_TVA"]]
+            taux_tva = taux_tva.replace(".", ",")  #  fix decimal separator different from titelive_things files.
+            title = row[COLUMN_INDICES["TITLE"]]
+
+            ineligibility_reason = get_ineligibility_reason(
+                gtl_id=gtl_id,
+                code_support=code_support,
+                id_lectorat=id_lectorat,
+                taux_tva=taux_tva,
+                title=title,
+                comment=comments_dict.get(ean, ""),
+            )
+
+            if ineligibility_reason:
+                if ean in product_whitelist_eans:
+                    whitelisted_eans.append(ean)
+                else:
+                    reasons[ineligibility_reason]["eans"].append(ean)
+                    reasons[ineligibility_reason]["total"] += 1
+                    rejected_eans.append(ean)
+                    emails += reject_inappropriate_products(ean, send_booking_cancellation_emails=False, dry=dry)
+
+        logger.info(
+            "%s whitelisted EANs found in CSV (out of %s in product_whitelist table): %s",
+            len(whitelisted_eans),
+            len(product_whitelist_eans),
+            whitelisted_eans,
+        )
+
+        output_files: list[str] = []
+
+        with open(OUT_ALL_REJECTED_EANS_FILE_PATH, "wb+") as all_rejected_eans_csv:
+            all_rejected_eans_csv.write("\n".join(rejected_eans).encode())
```

Comment:
(détail) Tu dois utiliser `.encode()` parce que le fichier est ouvert en mode binaire. Mais tu peux l'ouvrir en mode texte : 

```suggestion
        with open(OUT_ALL_REJECTED_EANS_FILE_PATH, "w+") as all_rejected_eans_csv:
            all_rejected_eans_csv.write("\n".join(rejected_eans))
```

Idem plus bas pour les autres fichiers dans lesquels tu écris.